src/merkle/merkle-objects.ts

```diff
@@ -232,8 +232,11 @@ export class IpfsLeafCompare implements CompareFunction<Candidate> {
 export class BloomMetadata implements MetadataFunction<Candidate, TreeMetadata> {
   generateMetadata(leaves: Array<Node<Candidate>>): TreeMetadata {
     const bloomFilterEntries = new Set<string>()
+    const streamIds = new Set<string>()
```

Comment:
this can just be an Array, we already shouldn't have duplicate streams in the tree